### ISSUE TITLE
System.ServiceModel > 4.5.0-preview1-26228-01 SOAP FAILS

### ISSUE THREAD

**Issue by cgountanis** [dotnet/wcf#2908]
Anything past 4.5.0-preview1-26228-01 on these listed when trying to utilize CORE WCF now fails with Ntlm errors when it was working perfectly on remote/non-domain systems. Behind the firewall HTTP calls. Just getting "The HTTP request is unauthorized with client authentication scheme 'Negotiate'. The authentication header received from the server was NTLM". I had to revert from 4.5.0 back to 4.5.0-preview1-26228-01 and it is working remote perfectly again.

System.ServiceModel.Duplex
System.ServiceModel.Http
System.ServiceModel.NetTcp
System.ServiceModel.Security

What changed? Is there a simple code change I can do?

```
#region Binding
// Init the basic http binding.
var basicHttpBinding = new BasicHttpBinding()
{
	MaxReceivedMessageSize = int.MaxValue,
	MaxBufferSize = int.MaxValue,
	ReaderQuotas = XmlDictionaryReaderQuotas.Max
};

// Set the binding security mode based on the url scheme.
if (new Uri(model.ReportServerUrl).Scheme.ToUpper() == "HTTPS")
{
	basicHttpBinding.Security.Mode = BasicHttpSecurityMode.Transport;
}
else
{
	basicHttpBinding.Security.Mode = BasicHttpSecurityMode.TransportCredentialOnly;
}

// Binding client security transport settings.
basicHttpBinding.Security.Transport.ClientCredentialType = HttpClientCredentialType.Windows;
#endregion

#region Soap Client
// Create the report execution service soap client.
var serviceSoapClient = new ReportExecutionServiceSoapClient(basicHttpBinding, new EndpointAddress($"{model.ReportServerUrl}/ReportExecution2005.asmx"));

// Credentials.
serviceSoapClient.ClientCredentials.Windows.AllowedImpersonationLevel = System.Security.Principal.TokenImpersonationLevel.Impersonation;
serviceSoapClient.ClientCredentials.Windows.ClientCredential = new NetworkCredential(model.Username, model.Password, model.Domain);
#endregion

#region Report
// Init the trusted user header.
var trustedUserHeader = new TrustedUserHeader();

// Load report and get response.
var loadReportResponse = await serviceSoapClient.LoadReportAsync(trustedUserHeader, model.ReportPath, null);

// Create execution header using load report response information.
var executionHeader = new ExecutionHeader { ExecutionID = loadReportResponse.executionInfo.ExecutionID };

// Parameters.
var setExecutionParametersResponse = await serviceSoapClient.SetExecutionParametersAsync(executionHeader, trustedUserHeader, model.Parameters.ToArray(), "en-us");

// Render request.
var renderRequest = new RenderRequest(executionHeader, trustedUserHeader, model.ReportFormat, null);

// Render response.
var renderResponse = await serviceSoapClient.RenderAsync(renderRequest);
#endregion

return renderResponse;
```

**Comment by mconnew** [comment]
We improved the security around how credentials are handled compared with earlier versions. We now only use the set credentials when the server asks to use the same authentication scheme as your client is configured to use. So if your client is configured to use authentication scheme Digest and the server asks for Basic auth (which the password can be recovered from trivially), we won't provide the credentials. The value of you HttpTransportBindingElement.AuthenticationScheme needs to be set to System.Net.AuthenticationSchemes.Ntlm. It looks like there might a bug where we don't set this automatically, so setting it manually on the HttpTransportBindingElement will work around this issue. I'll investigate further about any missing behavior with this being automatically set.

**Comment by cgountanis** [comment]
Based on the code above I would love to upgrade and have the functionality (SSRS Report Export) working on the newer versions like tje old as mentioned above. I am not a god in this tech, can you explain what I need to change to get "HttpTransportBindingElement.AuthenticationScheme needs to be set to System.Net.AuthenticationSchemes.Ntlm" set right, or can I wait for the next version if it is a bug?

Thanks @mconnew

**Comment by cgountanis** [comment]
Ok, this is strange. I know I tried this but I changed Windows to Ntlm and it worked in my above code example. I am rebooting the remote server IIS/SQL-SSRS and testing again. Crazy not sure why it did not work a couple days ago.

```
// Binding client security transport settings.
basicHttpBinding.Security.Transport.ClientCredentialType = HttpClientCredentialType.Ntlm;
```

**Comment by mconnew** [comment]
We haven't made any changes to the the release build package. Are you sure you didn't have the ClientCredentialType set as Windows?

**Comment by mconnew** [comment]
I just checked the code and we do set this correctly based on the ClientCredentialType so it doesn't look like we have a bug. So in summary your usage was slightly incorrect which worked previously due to lax enforcement of authentication scheme. We made a change to match the full framework to only provide credentials when the server is using the same authentication scheme as we are expecting which is what introduced a problem for you.  
I'm closing this issue as it looks like you have a solution.
